### PR TITLE
ENH: [`CI` / `Docker`]: Create a workflow to temporarly build docker images in case dockerfiles are modified

### PR DESCRIPTION
**Pull request by younesbelkada**
Hi !

I would like to enhance the current way of testing docker build workflow, right now I need to manually change the rules for triggering the build docker image job to test if a PR leads to potential failure on docker build. In this PR I propose to simply trigger a simple workflow that just builds (without pushing) docker images that are modified by a PR

cc @pacman100 @BenjaminBossan

### CODE REVIEW

**Comment by younesbelkada** [comment]
@glegendre01 do you think it's ok to move forward with this new workflow?

File: .github/workflows/test-docker-build.yml
Comment (younesbelkada) [inline .github/workflows/test-docker-build.yml:6]:
```suggestion
    # Run only when DockerFile files are modified
```

**Review by glegendre01 — CHANGES_REQUESTED** [review]

**Review by glegendre01 — APPROVED** [review]

**Review by BenjaminBossan — APPROVED** [review]
In general, LGTM, thanks for adding this useful feature.

A minor concern I have is about adding this new GH action. Could it be a possible security concern? Maybe it's a non-issue, it was just something that crossed my mind.

**Comment by younesbelkada** [comment]
Thanks for the review! 
I think that it should be all good as we don't push any docker image using that workflow and we use a GHA pinned on a specific commit (from @glegendre01 comment), in any case all PRs needs a maintainer approval before running the CI so we should be careful when reviewing PRs where we have some doubts (by pinging @glegendre01 if we have some strong doubts 🙏 )

**Comment by younesbelkada** [comment]
Also I need to merge this PR before properly testing it as GH does not allow to add new workflow from PRs :D

**Review by pacman100 — APPROVED** [review]
Thank you @younesbelkada, nice QoL update for contributors! 🤗

**Pull request merged**
